src/transforms/filter.jl

```diff
@@ -0,0 +1,42 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Filter(f, table)
+
+The transform that filters the rows based on a given function.
+"""
+struct Filter{F} <: Transform
+  pred::F
+end
+
+# Default filter to be used when no input function is given
+# Returns true if any element in the row evaluates to true
+Filter() = Filter(row -> any(row))
+
+isrevertible(::Type{Filter}) = true
+
+function apply(transform::Filter, input_table)
+  # Converting to Tables.rowtable to allow length function and indexing 
+  table = Tables.rowtable(input_table)
```

Comment:
Do we really need to convert to rowtable? Can't we just loop over the Tables.rows(table) iterator and check if the given row satisfies the predicate? If it does, then we can increment a counter and save the row, otherwise we just increment the counter.